src/util/address.rs

```diff
@@ -257,38 +258,38 @@ impl Address {
     pub fn p2sh(script: &script::Script, network: Network) -> Address {
         Address {
             network: network,
-            payload: Payload::ScriptHash(hash160::Hash::hash(&script[..])),
+            payload: Payload::ScriptHash(ScriptHash::hash(&script[..])),
         }
     }
 
     /// Create a witness pay to public key address from a public key
     /// This is the native segwit address type for an output redeemable with a single signature
     pub fn p2wpkh(pk: &key::PublicKey, network: Network) -> Address {
-        let mut hash_engine = hash160::Hash::engine();
+        let mut hash_engine = WPubkeyHash::engine();
```

Comment:
@apoelstra Is there really a semantic difference between a `PubkeyHash` and `WPubkeyHash`? Both are a hash of a pubkey. I'm not sure if it makes sense to change name based on the context the value is used in; IMO it should be based on the content they represent/hashes.

`WScriptHash` is different because it's sha256 instead of `ScriptHash`'s hash160.

Opinions?

Reply:
I don't have a strong opinion either way, but it seems reasonable to me to distinguish pubkeyhash and wpubkeyhash....at the very least so each can implement `Into<Address>` and do the right thing.

Reply:
IMO that's really a wrong way to create addresses. Plus, you need a network.